drv/lpc55-swd/src/main.rs

```diff
@@ -676,6 +724,26 @@ impl ServerImpl {
         }
     }
 
+    fn swd_dongle_detected(&self) -> bool {
+        cfg_if::cfg_if! {
+            if #[cfg(any(
+                target_board = "gimlet-b",
+                target_board = "gimlet-c"
+            ))] {
+                use drv_lpc55_gpio_api::*;
+
+                let gpio = Pins::from(self.gpio);
+
+                match gpio.read_val(SP_TO_ROT_JTAG_DETECT_L) {
+                    Value::Zero => true,
+                    Value::One => false,
+                }
```

Comment:
So earnest follow up: how strongly do you feel about this one?  To me, the first is easier to read, in part because this is already an active-low bit -- but maybe that's what makes the second easier to read to you?

Reply:
As _another_ alternative, you could do `matches!(gpio.read_val(SP_TO_ROT_JTAG_DETECT_L), Value::Zero)`, to preserve the mouth-feel of the `match` statement.

(I don't feel strongly about any of these three options; the current code is fine, just not what I would have written)

Reply:
Okay, that one is even less readable (to me!) -- which has me warming up to your first suggestion!